### PR TITLE
Add ContactPicker component

### PR DESCRIPTION
**Pull request by drazik**
:warning: Base branch should be changed to master after https://github.com/cozy/cozy-ui/pull/1273 has been merged.

Add a `ContactPicker` components that renders like a select, but opens a `ContactsListModal`. Also add the possibility to use it with a `Field` with `type="contact"`.

See https://drazik.github.io/cozy-ui/react/#!/ContactPicker and https://drazik.github.io/cozy-ui/react/#

### CODE REVIEW

File: stylus/components/forms.styl [hunk 1]
```diff
@@ -428,6 +428,7 @@ $select
     appearance none
     background embedurl('../../assets/icons/illus/bottom-select.svg') right rem(16) center no-repeat
     background-size rem(14)
+    text-align left
```
Comment (ptbrowne) [inline stylus/components/forms.styl]: Why was this necessary ? Because otherwise in a context where the text-align is different (center for example), we want to reset it ?

File: react/ContactPicker/index.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const { placeholder, onChange, selectedContact, ...rest } = props
+  const [showContactsList, setShowContactsList] = useState(false)
+
+  const handleChange = contact => {
+    onChange(contact)
+  }
+
+  return (
+    <>
+      <FakeSelect {...rest} onClick={() => setShowContactsList(true)}>
```
Comment (ptbrowne) [inline react/ContactPicker/index.jsx]: SelectControl instead of FakeSelect ?

File: stylus/components/forms.styl (hunk 1 above)
Comment (drazik) [reply stylus/components/forms.styl]: Because I'm using a `button` to render the fake select. And the default text align for buttons is center. But we always want the text to be aligned on the left by default when rendering something that looks like a select, no matter which underlying element is used.

File: react/ContactPicker/styles.styl [hunk 3]
```diff
@@ -0,0 +1,5 @@
+@require 'tools/mixins'
+@require '../../stylus/components/forms'
```
Comment (Crash--) [inline react/ContactPicker/styles.styl]: @require 'components/forms' should work?

File: react/Field/index.jsx [hunk 4]
```diff
@@ -211,6 +222,17 @@ const Field = props => {
...
+  'date',
+  'email',
+  'password',
+  'select',
+  'textarea',
+  'text',
+  'url',
+  'number'
+]
+
```
Comment (Crash--) [inline react/Field/index.jsx]: 👏

**Comment by drazik** [comment]
> I think it'd be nice to have a small test
>
>   click on the field
>   click on a person
>   check that the person's name is in the field ?
>
> What do you think ?

Yep, I'm going to add it.

> Also in the styleguidist, the Field seems very large ? Is it expected ?

Actually, it seems the Field is the same size as other ones, but the SelectBox is slightly smaller. I think we should align the SelectBox on other field types?

**Comment by ptbrowne** [comment]
>  I think we should align the SelectBox on other field types?

I am not sure but if they have two different custom sizes, I think it'd be ok yes to align one with the other.

**Comment by drazik** [comment]
It had a 30rem max-width where other control types had 32rem. I set it to 32rem too.

**Comment by ptbrowne** [comment]
Using a variable for both would be the best I guess, but LGTM anyway :+1: Thanks !
